.dotnet/src/Custom/Files/IncrementalUploadJob.cs

```diff
@@ -0,0 +1,101 @@
+using System;
+using System.ClientModel;
+using System.ClientModel.Primitives;
+
+namespace OpenAI.Files;
+
+/// <summary>
+/// A representation of a multi-part upload operation that can add data parts in parallel, useful for uploading very
+/// large files.
+/// </summary>
+/// <remarks>
+/// <see cref="IncrementalUploadJob"/> instances are obtained via
+/// <see cref="FileClient.CreateIncrementalUploadJob(BinaryContent, RequestOptions)"/> or rehydrated via
+/// <see cref="Rehydrate(OpenAI.Files.FileClient, ContinuationToken)"/> using a <see cref="RehydrationToken"/> from
+/// an existing instance.
+/// </remarks>
+public partial class IncrementalUploadJob : ClientResult
```

Comment:
I'm about to ramble a little bit wondering if there might be a more appropriate name for this class. 😄

The word "incremental" tends to have a positive connotation and makes it sound a little bit like uploading a file incrementally is a feature or like it is something desirable. I would argue that in this case this is closer to being an implementation detail than to being a feature. If I'm understanding Uploads correctly, the real feature is allowing users to upload larger files. This is achieved by having users break down the file into smaller parts and having them upload each part individually, which I assume improves reliability. And then, uploading parts can also be parallelized and done out of order, which improves time to completion. Both of these benefits are unrelated to the fact that the Upload can be done incrementally. Furthermore, when I think of "incremental", I think: "I can take my time and do it little by little", but according to the documentation, the Upload expires one hour after it was created. Does this mean that if the Upload is not completed before that, users have to start over? If so, presenting it as "incremental" could be misleading.

I'm also a little conflicted by the word "job". When I think of "job", I think of "a unit of work". But an Upload object in particular is not exactly doing any work, and in fact it won't do anything by itself. Instead, the user has to manually add parts one by one and even manually mark the upload as completed when they are done. The user has to do all the work. 😅

So I'm trying to think of alternatives... I like the simplicity of just calling it "FileUpload", but I recognize that it would be confusing given the existing `UploadFile` method in the `FileClient` that returns an `OpenAIFileInfo`. What about "MultipartFileUpload"? I think "multipart" has a more neutral connotation compared to "incremental" so I think it's less misleading. I'm also considering "SegmentedFileUpload", but we would need to call them "segments" instead of "parts", which deviates from the REST API.

What do you think?

P.S. More rambling: The funny thing is that, if it weren't for protocol methods and we had infinite time to work on this, I feel like this class wouldn't even need to exist. Take the existing `UploadFile` method that I mentioned above. Before sending the request, the `FileClient` could check the size of the file that the user is trying to upload. If it's a small file, it works like it works today, but if it's a file with a size past a certain threshold, the `FileClient` switches to using the Uploads API without the user having to know about it. We can split the file into parts on behalf of the user. We can even parallelize uploading each part ourselves so that the user doesn't have to worry about multiple threads. In the end, the user gets back an `OpenAIFileInfo` regardless of what file they tried to upload without having to think about it. Maybe we can write a convenience method like this in the future.

Reply:
I agree that "job" tends to mean a job running asynchronously on a service in the context of a cloud service - I would lean against it given that.  It might be useful to look at how the samples read to assist with naming here -- fileUpload.AddPart(...) and fileUpload.Complete(...) read quite well.

Reply:
I'm very much OK with renaming it -- I just want to be careful that we're not obfuscating things too much.

The `Upload` in this case is a service-persistent entity that buffers data segments and then assembles them into a file once it's complete; it's not quite the same as a fine-tuning job in that it still needs more input (including an explicit "I'm done") to finish, but I think it's important to give it *some sort of name hint* that clarifies it's segmented/distributed/partialized/resumable/parallelizable/etc.. Relying on the word "Upload" by itself doesn't capture enough of the "it's in parts, you have to tell it to complete, you have to order the parts yourself, ..." specifics.

How about `FilePartUpload`? `filePartUpload.AddPart(...)` is a *little* redundant, but it seems like a good middle ground that avoids the worst "underspecification" confusion that customers might have if they call "CreateUpload" and wonder why it's not a finished "Upload." "PartializedFileUpload" would be even more precise (clarifying it's not just uploading one part of something), but also super wordy.

Reply:
For the moment, I'll remove the "Job" and leave it as "IncrementalUpload."